lib/src/main/java/com/diffplug/spotless/java/PalantirJavaFormatStep.java

```diff
@@ -0,0 +1,152 @@
+
+	/** Creates a step which formats everything - code, import order, and unused imports. */
+	public static FormatterStep create(Provisioner provisioner) {
+		return create(defaultVersion(), provisioner);
+	}
+
+	/** Creates a step which formats everything - code, import order, and unused imports. */
+	public static FormatterStep create(String version, Provisioner provisioner) {
+		return create(version, DEFAULT_STYLE, provisioner);
+	}
+
+	/** Creates a step which formats everything - code, import order, and unused imports. */
+	public static FormatterStep create(String version, String style, Provisioner provisioner) {
+		return create(MAVEN_COORDINATE, version, style, provisioner);
+	}
+
+	/** Creates a step which formats everything - groupArtifact, code, import order, and unused imports. */
+	public static FormatterStep create(String groupArtifact, String version, String style, Provisioner provisioner) {
+		Objects.requireNonNull(groupArtifact, "groupArtifact");
+		if (groupArtifact.chars().filter(ch -> ch == ':').count() != 1) {
+			throw new IllegalArgumentException("groupArtifact must be in the form 'groupId:artifactId'");
+		}
+		Objects.requireNonNull(version, "version");
+		Objects.requireNonNull(style, "style");
+		Objects.requireNonNull(provisioner, "provisioner");
+		return FormatterStep.createLazy(NAME,
+				() -> new State(NAME, groupArtifact, version, style, provisioner),
+				State::createFormat);
+	}
+
+	static final Jvm.Support<String> JVM_SUPPORT = Jvm.<String> support(NAME).add(8, "1.1.0").add(11, "2.10.0");
+
+	public static String defaultGroupArtifact() {
+		return MAVEN_COORDINATE;
+	}
+
+	/** Get default formatter version */
+	public static String defaultVersion() {
+		return JVM_SUPPORT.getRecommendedFormatterVersion();
+	}
+
+	public static String defaultStyle() {
+		return DEFAULT_STYLE;
+	}
+
+	static final class State implements Serializable {
+		private static final long serialVersionUID = 1L;
+
+		/** The jar that contains the formatter. */
+		final JarState jarState;
+		final String stepName;
+		final String version;
+		final String style;
+
+		State(String stepName, String groupArtifact, String version, String style, Provisioner provisioner) throws Exception {
+			JVM_SUPPORT.assertFormatterSupported(version);
+			this.jarState = JarState.from(groupArtifact + ":" + version, provisioner);
+			this.stepName = stepName;
+			this.version = version;
+			this.style = style;
+		}
+
+		@SuppressWarnings({"unchecked", "rawtypes"})
+		FormatterFunc createFormat() throws Exception {
+			ClassLoader classLoader = jarState.getClassLoader();
+
+			// instantiate the formatter and get its format method
+			Class<?> optionsClass = classLoader.loadClass(OPTIONS_CLASS);
+			Class<?> optionsBuilderClass = classLoader.loadClass(OPTIONS_BUILDER_CLASS);
+			Method optionsBuilderMethod = optionsClass.getMethod(OPTIONS_BUILDER_METHOD);
+			Object optionsBuilder = optionsBuilderMethod.invoke(null);
```

Comment:
If you want, all this reflection can be replaced pretty easily by following one of the examples at #524.